lib/codegen/CreateTerm.cpp

```diff
@@ -199,15 +205,53 @@ llvm::Value *getBlockHeader(llvm::Module *Module, KOREDefinition *definition, co
   return llvm::ConstantStruct::get(BlockHeaderType, llvm::ConstantInt::get(llvm::Type::getInt64Ty(Module->getContext()), headerVal));
 }
 
-llvm::Value *allocateTerm(llvm::Type *AllocType, llvm::BasicBlock *block, const char *allocFn) {
-  return allocateTerm(AllocType, llvm::ConstantExpr::getSizeOf(AllocType), block, allocFn);
+static llvm::Value *addrspaceCast(llvm::Value *val, llvm::BasicBlock *block, int from, int to) {
+  auto AllocType = val->getType()->getPointerElementType();
+  bool hasUnnamed = false;
+  std::string name = "addrspace_" + std::to_string(from) + "_to_" + std::to_string(to) + "_" + getMangledTypeStr(AllocType, hasUnnamed);
+  auto addrspace = llvm::CallInst::Create(getOrInsertFunction(block->getModule(), name, llvm::PointerType::get(AllocType, to), llvm::PointerType::get(AllocType, from)), {val}, "", block);
+  return addrspace;
+}
+
+llvm::Value *addrspaceCast0to1(llvm::Value *val, llvm::BasicBlock *block) {
+  return addrspaceCast(val, block, 0, 1);
+}
+
+llvm::Value *addrspaceCast1to0(llvm::Value *val, llvm::BasicBlock *block) {
+  return addrspaceCast(val, block, 1, 0);
+}
+
+llvm::Value *ptrToInt(llvm::Value *val, llvm::BasicBlock *block) {
+  bool hasUnnamed = false;
+  std::string name = "ptrtoint_i64." + getMangledTypeStr(val->getType(), hasUnnamed);
+  auto ptr = llvm::CallInst::Create(getOrInsertFunction(block->getModule(), name, llvm::Type::getInt64Ty(block->getModule()->getContext()), val->getType()), {val}, "", block);
+  return ptr;
 }
```

Comment:
Why are these implemented by concatenating strings to get a function name - i.e. why don't we construct an `[AddrSpaceCastInst](https://llvm.org/doxygen/classllvm_1_1AddrSpaceCastInst.html)` (c.f. pointers to integers) directly?

Happy with this if there's a good reason for it, but getting access to LLVM intrinsics through string building seems a bit off to me.

Reply:
I'll add a comment explaining this.